#! /bin/sh /usr/share/dpatch/dpatch-run
## Description: add some description
## Origin/Author: add some origin or author
## Bug: bug URL
## 01_gtk_disable_deprecated.dpatch by Ralf Treinen <treinen@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: undefine GTK_DISABLE_DEPRECATED before including gtk/gtk.h.
## DP: patch taken from Ubuntu.

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' tiemu~/src/gui/calc/calc.c tiemu/src/gui/calc/calc.c
--- tiemu~/src/gui/calc/calc.c	2011-04-21 19:30:02.000000000 +0200
+++ tiemu/src/gui/calc/calc.c	2011-04-21 19:47:19.998760979 +0200
@@ -586,7 +586,7 @@
 
     // Install LCD refresh: 100 FPS (10 ms)
     tid = g_timeout_add((params.lcd_rate == -1) ? 50 : params.lcd_rate, 
-		(GtkFunction)hid_refresh, NULL);
+		(GSourceFunc)hid_refresh, NULL);
 
 	explicit_destroy = 0;
 	gtk_widget_show(main_wnd);	// show wnd here
@@ -640,7 +640,7 @@
 	g_source_remove(tid);
 
 	tid = g_timeout_add((params.lcd_rate == -1) ? 50 : params.lcd_rate, 
-		(GtkFunction)hid_refresh, NULL);
+		(GSourceFunc)hid_refresh, NULL);
 }
 
 int hid_switch_with_skin(void)
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' tiemu~/src/gui/calc/popup.c tiemu/src/gui/calc/popup.c
--- tiemu~/src/gui/calc/popup.c	2011-04-21 19:26:31.000000000 +0200
+++ tiemu/src/gui/calc/popup.c	2011-04-21 19:47:19.998760979 +0200
@@ -31,7 +31,9 @@
    This means GtkTooltipsData is now deprecated. */
 #undef GTK_DISABLE_DEPRECATED
 
+#undef GTK_DISABLE_DEPRECATED
 #include <gtk/gtk.h>
+#define GTK_DISABLE_DEPRECATED
 #include <glade/glade.h>
 #include <stdlib.h>
 #include <string.h>
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' tiemu~/src/gui/debugger/dbg_romcall.c tiemu/src/gui/debugger/dbg_romcall.c
--- tiemu~/src/gui/debugger/dbg_romcall.c	2011-04-21 19:26:31.000000000 +0200
+++ tiemu/src/gui/debugger/dbg_romcall.c	2011-04-21 19:48:04.506429940 +0200
@@ -29,7 +29,9 @@
 #  include <config.h>
 #endif
 
-#include <gtk/gtk.h>
+#undef GTK_DISABLE_DEPRECATED
+ #include <gtk/gtk.h>
+#define GTK_DISABLE_DEPRECATED
 #include <glade/glade.h>
 #include <string.h>
 
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' tiemu~/src/gui/filesel.c tiemu/src/gui/filesel.c
--- tiemu~/src/gui/filesel.c	2011-04-21 19:26:31.000000000 +0200
+++ tiemu/src/gui/filesel.c	2011-04-21 19:47:19.998760979 +0200
@@ -41,7 +41,9 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#undef GTK_DISABLE_DEPRECATED
 #include <gtk/gtk.h>
+#define GTK_DISABLE_DEPRECATED
 #include <string.h>
 
 #ifdef __WIN32__
